addons/web/static/src/xml/dialog.xml

```diff
@@ -3,11 +3,16 @@
 
 <!-- These templates are accessible in backend and frontend -->
 
-<div t-name="Dialog" t-attf-class="modal#{technical ? ' o_technical_modal' : ''}" tabindex="-1" data-backdrop="static" role="dialog" aria-hidden="true" t-att-id="_.uniqueId('modal_')">
+<div t-name="Dialog" t-attf-class="modal#{technical ? ' o_technical_modal' : ''} #{isMobile ? 'o_modal_full': ''}" tabindex="-1" data-backdrop="static" role="dialog" aria-hidden="true" t-att-id="_.uniqueId('modal_')">
     <div class="modal-dialog">
         <div class="modal-content">
             <div class="modal-header">
-                <button type="button" class="close" data-dismiss="modal" aria-hidden="true">×</button>
+                <t t-if="isMobile">
+                    <i class="fa fa-arrow-left o_btn_back mr8" data-dismiss="modal" aria-hidden="true"/>
+                </t>
+                <t t-else="">
+                    <button type="button" class="close" data-dismiss="modal" aria-hidden="true">×</button>
+                </t>
```

Comment:
If you create a new FullScreenDialog class (I'm not saying you should, but you might), then all this could become overrides on the original template right ?

Reply:
If we create a new FullScreenDialog, code will be redundant (like binding methods on buttons, open and destroy overrides etc). Also after that, few of the checks have to be done within FormViewDialog(like whether to display remove button or not, and if yes, where to display it). So may be we shouldn't create new class.

Reply:
I don't understand why the code would be redundant. The idea would be to extend the current dialog class, not to create a brand new one, so everything that is already done in the normal dialog class would also be done in the FullScreen one, without duplication. Then after that, the code that is specific to the full screen one could run and do the additional stuffs.